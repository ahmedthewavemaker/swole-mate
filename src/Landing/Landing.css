*{
    box-sizing: border-box;
    
  }
  
  html{
      height: 100%;
  }
  .App {
    text-align: center;
  
    background-image: url('../images/dumbells.jpg');
    background-repeat: no-repeat ;
    background-position: center;
 
  }

  body{
    background-color: black;

  
  }
  
  .topList{
    text-decoration: none;
  }
  
  .intro{
    color: gold;
    font-family:'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    font-weight: bold;
    font-size: 24px;
    text-align: center;
    font-style: italic;
  }

  .wrapper{
    text-align: center;
    width: 800px;
    margin: auto;
    background-color: black;
    padding: 10px 4px;
    
 
  }

  section{
    font-weight: bolder;
  }

  h2, section{
    color: gold;
    font-size: 30px;

  }

  h3{
    color: gold;
    font-style: italic;
  }

  .form{
    color:gold;
    font-size: 22px;
    font-weight: bolder;
  }