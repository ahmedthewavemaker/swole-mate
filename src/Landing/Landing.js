import React, { Component } from 'react';
import './Landing.css';
import AppContext from '../AppContext'
import { render } from '@testing-library/react';
import dumbells from '../images/dumbells.jpg'



class Landing extends Component {
    static contextType = AppContext

    getWorkouts = (event) => {
        event.preventDefault();

        this.context.setEmail(event.target.email.value)
        this.props.history.push('/muscle')
    }

    render() {


        return (

            <div className="wrapper">

                <article className="intro"> Ever go the the gym and feel like you are missing a workout? <br/> Ever go to the gym and forget the routine you are supposed to implement that day?<br/> Don't you wish you had a checklist that you can just look at instead of trying to remember which workouts you need to do as other uninvited thoughts rush your head?<br/>  </article>

                <h2>YOU ARE NOT ALONE! </h2> <section> Presenting the Swole Mate! Your personal side-kick/guide that ensures you never skip a workout again! </section>

                {/* <img src={dumbells} alt='dumbells' /> */}

                <h3>Sign up now to get swole!</h3>
                <form onSubmit={this.getWorkouts}><div>

                    <label className="form" htmlFor="email">Email: </label>
                    <input placeholder="email@website.com" type="text" name="email" id="email" />
                    <button className='buttonz'>Send</button>
                </div></form>
            </div>
        );
    }
}

export default Landing;