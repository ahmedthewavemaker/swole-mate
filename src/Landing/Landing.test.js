import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Landing from './Landing';
import { MemoryRouter } from 'react-router-dom';

describe('Landing component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter> <Landing />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter> <Landing />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


});
