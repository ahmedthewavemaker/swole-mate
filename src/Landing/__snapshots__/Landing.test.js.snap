// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="wrapper"
  >
    <article
      className="intro"
    >
       Ever go the the gym and feel like you are missing a workout? 
      <br />
       Ever go to the gym and forget the routine you are supposed to implement that day?
      <br />
       Don't you wish you had a checklist that you can just look at instead of trying to remember which workouts you need to do as other uninvited thoughts rush your head?
      <br />
        
    </article>
    <h2>
      YOU ARE NOT ALONE! 
    </h2>
     
    <section>
       Presenting the Swole Mate! Your personal side-kick/guide that ensures you never skip a workout again! 
    </section>
    <h3>
      Sign up now to get swole!
    </h3>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          className="form"
          htmlFor="email"
        >
          email: 
        </label>
        <input
          id="email"
          name="email"
          placeholder="email@website.com"
          type="text"
        />
        <button>
          Send
        </button>
      </div>
    </form>
  </div>,
]
`;
